### PR TITLE
Properly namespace slice and map types

### PR DESCRIPTION
**Pull request by skmcgrail**
Fixes the namespace for member shapes that are used in maps or slices.

### CODE REVIEW

File: codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java [hunk 1]
```diff
@@ -269,10 +269,23 @@ public String apply(Object type, String indent) {
             if (type instanceof Symbol) {
                 Symbol typeSymbol = (Symbol) type;
                 addUseImports(typeSymbol);
-                if (isExternalNamespace(typeSymbol.getNamespace())) {
-                    return formatWithNamespace(typeSymbol);
+
+                Boolean isUniverseType = typeSymbol.getProperty(SymbolUtils.GO_UNIVERSE_TYPE, Boolean.class)
+                        .orElse(false);
```
Comment (jasdel) [inline codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java:274]: is `UniverseType` for builtins?

Comment (skmcgrail) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java:274]: Yes, the naming choice of "universe" comes from the Go specification https://golang.org/ref/spec#Predeclared_identifiers

Comment (JordonPhillips) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java:274]: Could you put that in a comment

File: codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java [hunk 2]
```diff
@@ -269,10 +269,23 @@ public String apply(Object type, String indent) {
             if (type instanceof Symbol) {
                 Symbol typeSymbol = (Symbol) type;
                 addUseImports(typeSymbol);
-                if (isExternalNamespace(typeSymbol.getNamespace())) {
-                    return formatWithNamespace(typeSymbol);
+
+                Boolean isUniverseType = typeSymbol.getProperty(SymbolUtils.GO_UNIVERSE_TYPE, Boolean.class)
```
Comment (JordonPhillips) [inline codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java:273]: Doesn't matter, but this could probably be `boolean`

Comment (skmcgrail) [reply codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java]: Got this cleaned up as well.

**Pull request merged**